Source/Scene/Expression.js

```diff
@@ -726,42 +782,101 @@ define([
             } else {
-                Color.fromCssColorString(args[0].evaluate(frameState, feature, result), result);
+                Color.fromCssColorString(args[0].evaluate(frameState, feature), result);
             }
         } else if (this._value === 'rgb') {
             Color.fromBytes(
-                args[0].evaluate(frameState, feature, result),
-                args[1].evaluate(frameState, feature, result),
-                args[2].evaluate(frameState, feature, result),
+                args[0].evaluate(frameState, feature),
+                args[1].evaluate(frameState, feature),
+                args[2].evaluate(frameState, feature),
                 255, result);
         } else if (this._value === 'rgba') {
             // convert between css alpha (0 to 1) and cesium alpha (0 to 255)
-            var a = args[3].evaluate(frameState, feature, result) * 255;
+            var a = args[3].evaluate(frameState, feature) * 255;
             Color.fromBytes(
-                args[0].evaluate(frameState, feature, result),
-                args[1].evaluate(frameState, feature, result),
-                args[2].evaluate(frameState, feature, result),
+                args[0].evaluate(frameState, feature),
+                args[1].evaluate(frameState, feature),
+                args[2].evaluate(frameState, feature),
                 a, result);
         } else if (this._value === 'hsl') {
             Color.fromHsl(
-                args[0].evaluate(frameState, feature, result),
-                args[1].evaluate(frameState, feature, result),
-                args[2].evaluate(frameState, feature, result),
+                args[0].evaluate(frameState, feature),
+                args[1].evaluate(frameState, feature),
+                args[2].evaluate(frameState, feature),
                 1.0, result);
         } else if (this._value === 'hsla') {
             Color.fromHsl(
-                args[0].evaluate(frameState, feature, result),
-                args[1].evaluate(frameState, feature, result),
-                args[2].evaluate(frameState, feature, result),
-                args[3].evaluate(frameState, feature, result),
+                args[0].evaluate(frameState, feature),
+                args[1].evaluate(frameState, feature),
+                args[2].evaluate(frameState, feature),
+                args[3].evaluate(frameState, feature),
                 result);
         }
         return result;
     };
 
+    Node.prototype._evaluateLiteralVector = function(frameState, feature) {
+        // Gather the components that make up the vector, which includes components from interior vectors.
+        // For example vec3(1, 2, 3) or vec3(vec2(1, 2), 3) are both valid.
+        //
+        // If the number of components does not equal the vector's size, then a DeveloperError is thrown - with two exceptions:
+        // 1. A vector may be constructed from a larger vector and drop the extra components.
```

Comment:
Is (1) also true in GLSL?

If not, is it hard/slow to implement the same behavior here?